### PR TITLE
feat(Schema): disable config - allow disabling schema task

### PR DESCRIPTION
**Pull request by luvkapur**
This PR adds a new config for `teambit.semantics/schema` aspect called `disabled` - which disables the extraction of schema during the build task and also disables extracting the schema ad hoc when queried from the workspace.

### CODE REVIEW

File: scopes/semantics/schema/schema.task.ts [hunk 1]
```diff
@@ -33,6 +33,8 @@ export class SchemaTask implements BuildTask {
     const rootDir = context.capsuleNetwork.capsulesRootDir;
     await pMapSeries(capsules, async (capsule) => {
       const component = capsule.component;
+      const isSchemaTaskDisabled = component.state.aspects.get(this.aspectId)?.data?.disabled;
```
Comment (GiladShoham) [inline scopes/semantics/schema/schema.task.ts:36]: Better to have an api in the schema main runtime that calculate that data then call it from the task

**Review by GiladShoham — CHANGES_REQUESTED** [review]

**Review by davidfirst — APPROVED** [review]

File: scopes/semantics/schema/schema.task.ts (hunk 1 above)
Comment (davidfirst) [reply scopes/semantics/schema/schema.task.ts:36]: @luvkapur , I guess what @GiladShoham  wants is a method `getData()` in `schema.main` which returns `component.state.aspects.get(this.aspectId)?.data`

Comment (luvkapur) [reply scopes/semantics/schema/schema.task.ts]: @davidfirst @GiladShoham I added 
```
 isSchemaTaskDisabled(component: Component) {
    return component.state.aspects.get(SchemaAspect.id)?.data?.disabled;
  }
  ```
  to `schema.main`
  
  and invoke it from the task like this
  `const isTaskDisabled = this.schema.isSchemaTaskDisabled(component);`

File: scopes/semantics/schema/schema.main.runtime.ts [hunk 2]
```diff
@@ -171,6 +180,16 @@ export class SchemaMain {
     return this;
   }
 
+  async getSchemaData(): Promise<{ disabled?: boolean }> {
```
Comment (GiladShoham) [inline scopes/semantics/schema/schema.main.runtime.ts:183]: Rename from calc to get

**Review by GiladShoham — CHANGES_REQUESTED** [review]

**Review by GiladShoham — APPROVED** [review]

**Pull request merged**